bundles/org.eclipse.equinox.registry/src/org/eclipse/core/internal/registry/TableReader.java

```diff
@@ -299,7 +299,7 @@ private String[] readPropertiesAndValue(DataInputStream inputStream) throws IOEx
 			return RegistryObjectManager.EMPTY_STRING_ARRAY;
 		String[] properties = new String[numberOfProperties];
 		for (int i = 0; i < numberOfProperties; i++) {
-			properties[i] = readStringOrNull(inputStream);
+			properties[i] = readStringOrNull(inputStream).intern();
```

Comment:
The method name suggest it could be null, maybe better intern in the method itself?

Reply:
thanks for the hint - in which method?

Reply:
> thanks for the hint - in which method?

`readStringOrNull`

Reply:
I'm not sure if there are two such methods:

![image](https://github.com/eclipse-equinox/equinox/assets/208716/0d11c344-44e3-4d11-a77b-c36b91ccf29d)

Reply:
> I'm not sure if there are two such methods:
> 
> ![image](https://private-user-images.githubusercontent.com/208716/306647687-0d11c344-44e3-4d11-a77b-c36b91ccf29d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1MjQ2NDksIm5iZiI6MTcwODUyNDM0OSwicGF0aCI6Ii8yMDg3MTYvMzA2NjQ3Njg3LTBkMTFjMzQ0LTQ0ZTMtNGQxMS1hNzdiLWMzNmI5MWNjZjI5ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyMVQxNDA1NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZjI4N2IyZDQ3NTRkZWZkY2ZhZWI1ZDhmYWI3OTg0MzNhNGVhMWY3ZjhhMWEzZTkwODRjYjgzY2JkZDkyNjNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.hBDDqBGK-EmTMz71yv92GCDO-Hu_oGMOO_bbav_AJ4E)

Well the one that this TabelReader calls then...